src/configureAWS.ts

```diff
@@ -20,51 +20,49 @@ import * as aws from 'aws-sdk';
 import {logger} from './logger';
 import {AWSRegion} from './aws-regions';
 
-async function loadSharedIniFile(profile?: string) {
+function getCredentialsProviderChain(profile?: string) {
+  const hasDotAWS = (awsUserDir && fs.existsSync(awsUserDir));
+  const providers = [];
+  if (profile) {
+    if (profile.startsWith('arn:')) {
+      throw new Error(
+        'The impossible happened. getCredentialsProviderChain was passed an arn formatted profile');
+      // NOTE: arn formatted profiles are handled elsewhere
+    }
+    if (!hasDotAWS) {
+      throw new Error('AWS profile provided but ~/.aws/{config,credentials} not found.');
+    }
+    providers.push(() => new aws.SharedIniFileCredentials({profile}));
+  } else {
+    providers.push(() => new aws.EnvironmentCredentials('AWS'));
```

Comment:
I think this branch should just do `providers.concat(aws.CredentialProviderChain.defaultProviders)` rather than essentially recreating that behaviour checking `hasDotAWS` (the chain will skip over that provider if the file isn't present), as well as missing a number of providers.

Reply:
How would we thread the `profile` argument through to the `defaultProviders `?